### ISSUE TITLE
Running install_receptor.yml fails when adding execution node(Rocky Linux 9.0).

### ISSUE THREAD

**Issue by shihou-ono** [ansible/awx#13287]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Bug Summary

I prepared a Rocky Linux 9.0 machine for the execution node of the AWX.  
And I ran install_receptor.yml according to [execution_nodes.md](https://github.com/ansible/awx/blob/e6497be200d3311288237859a351483f769c0fe7/docs/execution_nodes.md) but it fails at the job "Install dependencies specific to the node type"

The workaround for this problem is to modify [install_bundle/group_vars/all.yml](https://github.com/ansible/awx/blob/devel/awx/api/templates/instance_install_bundle/group_vars/all.yml) like following.

```yaml
receptor_dependencies:
  # - python39-pip
  - python3-pip
```

### AWX version

21.7

### Select the relevant components

- [ ] UI
- [X] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Installation method

kubernetes

### Modifications

no

### Operating system

Rocky Linux 9.0

### Steps to reproduce

1. Setup a Rocky Linux 9.0 machine for the execution node of the AWX.
1. According to [execution_nodes.md](https://github.com/ansible/awx/blob/e6497be200d3311288237859a351483f769c0fe7/docs/execution_nodes.md) overview, perform steps 1 to 3.
1. At the step 4,
    1. Run `ansible-galaxy collection install -r requirements.yml`
    1. Modify inventory.yml correctly.
    1. Run `ansible-playbook -i inventory.yml install_receptor.yml`

### Expected results

Running playbook ends successfully.

### Actual results

Failed at the job "Install dependencies specific to the node type" with following error msg.

```json
fatal: [remote-execution]: FAILED! => {
     "changed": false,
     "failures": [
         "No package python39-pip available."
     ],
     "invocation": {
         "module_args": {
             "allow_downgrade": false,
             "allowerasing": false,
             "autoremove": false,
             "bugfix": false,
             "cacheonly": false,
             "conf_file": null,
             "disable_excludes": null,
             "disable_gpg_check": false,
             "disable_plugin": [],
             "disablerepo": [],
             "download_dir": null,
             "download_only": false,
             "enable_plugin": [],
             "enablerepo": [],
             "exclude": [],
             "install_repoquery": true,
             "install_weak_deps": true,
             "installroot": "/",
             "list": null,
             "lock_timeout": 30,
             "name": [
                 "python39-pip"
             ],
             "nobest": false,
             "releasever": null,
             "security": false,
             "skip_broken": false,
             "state": "present",
             "update_cache": false,
             "update_only": false,
             "validate_certs": true
         }
     },
     "msg": "Failed to install some of the specified packages",
     "rc": 1,
     "results": []
}
```

### Additional information

[install_receptor.yml](https://github.com/ansible/awx/blob/devel/awx/api/templates/instance_install_bundle/install_receptor.yml) calls [setup-RedHat.yml](https://github.com/ansible/receptor-collection/blob/main/roles/setup/tasks/setup-RedHat.yml) via the role of [ansible.receptor.setup](https://github.com/ansible/receptor-collection/blob/main/roles/setup/tasks/main.yml)  
And it fails at the job "Install dependencies specific to the node type"

```yaml
setup-RedHat.yml
---
- name: Install receptor packages
   ansible.builtin.dnf:
     name: "{{ receptor_packages }}"
     state: present

- name: Install dependencies specific to the node type  # <=this
   ansible.builtin.dnf:
     name: "{{ receptor_dependencies | default([]) }}"
     state: present
```

The variable `receptor_dependencies` is defined at install_bundle/group_vars/all.yml downloaded form the AWX and it comes from this template [all.yaml](https://github.com/ansible/awx/blob/devel/awx/api/templates/instance_install_bundle/group_vars/all.yml)

```yaml
all.yml

receptor_user: awx
receptor_group: awx
...snip
receptor_dependencies:  # <=this
  - python39-pip

podman_user: "{{ receptor_user }}"
podman_group: "{{ receptor_group }}"
```

Actually, on Rocky 9.0(and may be Rhel 9 and its clones) `dnf install python39-pip` doesn't work but `dnf install python3-pip` works (and default Python version is 3.9)

So, the workaround is to modify all.yml like following.

```yaml
receptor_dependencies:
  # - python39-pip
  - python3-pip
```

I understand this issue dependents specific environment, so I don't mind if this report is regarded not as a bug report, but as an introduction to workaround.  
I hope this report helps my fellow not spending time to struggle with this problem.

**Comment by kurokobo** [comment]
Good catch, this problem was once reported in the awx-operator repository, but it seems to have been closed by the author and I totally forgot about that...: ansible/awx-operator#1090

**Comment by shihou-ono** [comment]
@kurokobo 

Thanks for your comment.

I confirmed this works on my environment

```yaml
all.yml

receptor_dependencies:
   "{{ ansible_distribution_version is version(9.0, '<') | ternary('python39-pip', 'python3-pip') }}"
```

```yaml
---
- hosts: all
   become: yes
   tasks:
     - name: debug ansible_distribution_version
       debug:
         msg: "{{ ansible_distribution_version }}"

         # returns "9.0"

     - name: debug receptor_dependencies
       debug:
         msg: "{{ receptor_dependencies }}"

         # returns "python3-pip"

```

But, I've not research whether it works on CentOS Stream 8 and RHEL 8.  
And there are better codes than mine, I think!!

**Comment by kurokobo** [comment]
@shihou-ono 
Hi, thanks for providing a solution😃
F.Y.I. CentOS and Fedora returns not `X.Y` but just `X` like `8`, `9`, or `38` as `ansible_distribution_version`,

```bash
TASK [debug] *******************************************************************
ok: [localhost] => {
    "dump": {
        "ansible_distribution": "CentOS",
        "ansible_distribution_major_version": "9",
        "ansible_distribution_release": "Stream",
        "ansible_distribution_version": "9"     👈👈👈
    }
}
```

```bash
TASK [debug] *******************************************************************
ok: [localhost] => {
    "dump": {
        "ansible_distribution": "Rocky",
        "ansible_distribution_major_version": "9",
        "ansible_distribution_release": "Blue Onyx",
        "ansible_distribution_version": "9.0"     👈👈👈
    }
}
```

and the result of `version(9.0, '<')` with it may be counterintuitive in some cases.

```yaml
---
- hosts: localhost
  vars:
    dump:
      8_0_lt_9_0: "{{ '8.0' is version(9.0, '<') }}"
      8_1_lt_9_0: "{{ '8.1' is version(9.0, '<') }}"
      8___lt_9_0: "{{ '8'   is version(9.0, '<') }}"
      9_0_lt_9_0: "{{ '9.0' is version(9.0, '<') }}"
      9_1_lt_9_0: "{{ '9.1' is version(9.0, '<') }}"
      9___lt_9_0: "{{ '9'   is version(9.0, '<') }}"     👈👈👈
  tasks:
    - debug:
        var: dump
```

```bash
$ ansible-playbook test.yml 
...
TASK [debug] *******************************************************************************************
ok: [localhost] => {
    "dump": {
        "8_0_lt_9_0": true,
        "8_1_lt_9_0": true,
        "8___lt_9_0": true,
        "9_0_lt_9_0": false,
        "9_1_lt_9_0": false,
        "9___lt_9_0": true     👈👈👈
    }
}
```

Comparing to `9` instead of `9.0`, or use `ansible_distribution_major_version` instead can be a safer solution.

If you are interested, you could try making a PR for this issue😃

**Comment by shihou-ono** [comment]
@kurokobo 

Oh, I didn't suppose this `9___lt_9_0` case at all.
This is really important information for me, Thanks!!

I'll replace `ansible_distribution_version` => `ansible_distribution_major_version` when I send PR.

**Comment by relrod** [comment]
@shihou-ono Thanks in advance for the PR! :tada:

**Comment by relrod** [comment]
Just to add: This reproduces on Fedora 37 too (and maybe earlier, haven't checked).

**Comment by shihou-ono** [comment]
@relrod 

Thanks for your information!!!
Now I checked on Fedora 28(base of RHEL8) & Fedora34(base of RHEL9) .
And following worked.

```yaml
receptor_dependencies:
   - "{{ ansible_distribution_major_version is version(9.0, '<') | ternary('python39-pip', 'python3-pip') }}"
```

**Comment by kurokobo** [comment]
The default value for `receptor_dependencies` will be updated to `python3-pip` in #14088.
So RHEL9 based installation will be succeeded without any issue, but as a side effect, I guess RHEL8 based installation will be failed due to new default value.

**Comment by kurokobo** [comment]
The task to install `python3-pip` (or `python39-pip` for RHEL8`) will be added to the collection side: https://github.com/ansible/receptor-collection/pull/47

The variable `receptor_dependencies` has been removed from AWX side (https://github.com/ansible/awx/pull/13904) in feature branch, so this issue will be able to be closed after https://github.com/ansible/receptor-collection/pull/47 and the release of hop nodes.
